### PR TITLE
Convert 'echo' command to a manufacturer specific 'ping' command

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

The `echo` command is an hand rolled command. It has been super useful but currently, in its current shape, it is a pain as it make some part of the code harder to write while it will never really be used as it is in an end product.

This PR replace `echo` by a `ping` manufacturer specific command. This is nice because if add supports for some part of the code that where not used until then, notably the manufacturer specific side of ZCL clusters, and it removes the need for some special handling support in various places.

The changes in `examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h` has been done using `AppBuilder` while the rest of the `gen/` changes has been done using `ZAP`.

 #### Summary of Changes
 * Add a manufacturer specific command to the `all-clusters-app` configuration
 * Update `src/app/zap-templates/call-command-handler-src.zapt` to support manufacturer specific command
 * Remove `echo` command support in `examples/common/chip-app-server`
 * Simplify `chip-tool` by killing the `EchoCommand` class
 * Simplify `chip-tool` by mergin the `NetworkCommand` and `ModelCommand` classes together

### CODE REVIEW

**Review by rwalker-apple — APPROVED** [review]

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.zap [hunk 1]
```diff
@@ -2822,4 +2830,4 @@
       "networkId": 1
     }
   ]
-}
+}
```
Comment (rwalker-apple) [inline examples/all-clusters-app/all-clusters-common/all-clusters-app.zap:2833]: nit: can we keep newlines at EOF to eliminate this kind of noise in PRs?

**Review by andy31415 — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Comment by vivien-apple** [comment]
Pushed an update to resolve merge conflicts.

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.zap (hunk 1 above)
Comment (mspang) [reply examples/all-clusters-app/all-clusters-common/all-clusters-app.zap:2833]: Does restyler-whitespace fix this, so we could just run that on .zap files?

**Review by mspang — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@vivien-apple Please rebase to tip to retrigger CI on top of the pigweed fixes?

File: examples/chip-tool/commands/clusters/ModelCommand.cpp [hunk 2]
```diff
@@ -77,7 +154,24 @@ bool ModelCommand::Decode(PacketBufferHandle & buffer) const
...
+    SetCommandExitStatus(success);
+    UpdateWaitForResponse(false);
+}
+
+void ModelCommand::OnStatusChange(void)
+{
+    ChipLogProgress(chipTool, "DeviceStatusDelegate::OnStatusChange");
+}
+
+void ModelCommand::PrintBuffer(PacketBufferHandle & buffer) const
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/clusters/ModelCommand.cpp:167]: `const PacketBufferHandle &`, please.

**Review by bzbarsky-apple — COMMENTED** [review]
This just sends back a status response, unlike the existing echo command, which responds with the incoming data, right?  Why that change?

**Comment by vivien-apple** [comment]
> This just sends back a status response, unlike the existing echo command, which responds with the incoming data, right? Why that change?

You are right.
There are 2 reasons why I did that:
 1. The first one is that there is already an `echo` protocol now. I'm not saying I'm happy with that but since it is there it would not make sense to duplicate it.
 2. The second is because I felt like the goal of `echo` at the beginning was to make sure the data is encoded/decoded properly. Since `ping` will be encrypted/decrypted already, and since the command will be parse on both ends, it just seems redundant.

**Pull request merged**